src/jmclient/wallet.py

```diff
@@ -606,12 +618,20 @@ def get_key(self, mixdepth, address_type, index):
         raise NotImplementedError()
 
     def get_addr(self, mixdepth, address_type, index):
-        script = self.get_script(mixdepth, address_type, index)
-        return self.script_to_addr(script)
+        path = self.get_path(mixdepth, address_type, index)
+        return self.get_address_from_path(path)
 
     def get_address_from_path(self, path):
-        script = self.get_script_from_path(path)
-        return self.script_to_addr(script)
+        cache = self._get_cache_for_path(path)
+        addr = cache.get(b'A')
```

Comment:
is there a reason for choosing b'A', b'S' instead of more readable 'address', 'script'?

Reply:
Making that change would add roughly 165 kB to the size of my wallet file for no useful reason. It's a waste of disk space, and every byte takes time to handle. I would have used numeric keys, but Bencoding doesn't support that. I did consider having the cache object at each leaf in the tree be a list rather than a dict, but Bencoding does not support null values in lists. Overall a pretty feature-poor encoding format, TBH. Any reason you didn't go with standard pickling?

Reply:
I don't remember what @undeath said about this choice, afraid I didn't ask. Pickling is a bit dubious from the security point of view (at least, I remember reading about cases in which it introduced attack vectors), but I guess that only applies to cases like transferring serializations over the wire between untrusted participants. I *do* prefer less flexible serialization routines myself for that vague "flexible serialization of python objects feels less secure" reason.

But "vague", it is not some strongly held or carefully considered opinion.

Re: size: that's a good point, and certainly enough of a reason.